File: agent/app/models/edit.py
from pydantic import BaseModel, Field
from typing import List, Optional

class EditRequest(BaseModel):
    instruction: str = Field(..., min_length=1)
    file_path: str = Field(..., min_length=1)
    original_text: str = Field(..., min_length=1)
    user_context: Optional[str] = None
    
class EditResponse(BaseModel):
    file_path: str
    unified_diff: str
    updated_text: str
    warnings: List[str] = Field(default_factory=list)

File: extension/src/api/edit.ts
import { agentHttpClient } from "./client";

export type EditRequest = {
    instruction: string;
    file_path: string;
    original_text: string;
    user_context?: string | null;
};

export type EditResponse = {
    file_path: string;
    unified_diff: string;
    updated_text: string;
    warnings: string[];
};

export async function editCode(req: EditRequest): Promise<EditResponse> {
    const client = agentHttpClient();
    const resp = await client.post("/edit", req);
    return resp.data as EditResponse;
}
